src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -194,6 +195,12 @@ public function load(array $configs, ContainerBuilder $container)
             }
         }
 
+        // If the slugger is used but the String component is not available, we should throw an error
+        if (!class_exists(SluggerInterface::class)) {
+            $container->register('slugger', 'stdClass')
+                ->addError('You cannot use the Slugger since the String component is not installed. Try running "composer require symfony/string".');
```

Comment:
Do we actually use this pattern anywhere else ?

Reply:
I don't think so, because most of the time we have bundle configuration to help us determine whether the feature is used or not. In our case, typehinting is the only usage information we have, thus this definition.